test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -106,22 +123,36 @@ private void show(MouseEvent e) {
     }
 
     private static void openPopup(final Frame frame) throws Exception {
-        Robot robot = new Robot();
-        robot.setAutoDelay(200);
         robot.waitForIdle();
         Point pt = frame.getLocationOnScreen();
-        robot.mouseMove(pt.x + frame.getWidth() / 2, pt.y + 50);
+        int x = pt.x + frame.getWidth() / 2;
+        int y = pt.y + 50;
+        robot.mouseMove(x, y);
         robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
-        int x = pt.x + frame.getWidth() / 2;
-        int y = pt.y + 130;
+        robot.waitForIdle();
```

Comment:
When testing locally on my macOS machine, this waitForIdle seems to be the culprit for why this test takes much longer per iteration. Maybe worth looking into why?

Reply:
Replaced robot.waitForIdle with robot.delay(200). I will check the possibility of raising a new bug.

Reply:
Since no one expected for `robot.waitForIdle` to make the test so slow here, I suggest submitting a bug for evaluation. You have a working test case for this. The frame doesn't need to move across the screen.

Reply:
I raised a new bug for this issue JDK-8299937